comet/manager.py

```diff
@@ -179,13 +188,28 @@ def register_start(self, start_time, version, config=None):
                     "The broker is asking for state {} when state {} (start) was "
                     "registered.".format(reply.get("hash"), state_id)
                 )
-            self._send_state(state)
+            self._send_state(start_state)
 
-        self.states[state_id] = state
-        self.start_state = state_id
+        self.states[state_id] = start_state
+        self.start_state = start_state
         self.state_reg_time[state_id] = datetime.datetime.utcnow()
 
-        return
+        if register_datasets:
+            if config:
+                config_ds = self.register_dataset(
+                    config_state, None, config_state.state_type, True
+                )
+                return self.register_dataset(
+                    start_state, config_ds, start_state.state_type
+                )
+            # Todo: deprecated
+            # If there's no config, register the start state with a root dataset.
+            else:
+                return self.register_dataset(
+                    start_state, None, start_state.state_type, True
+                )
+        else:
```

Comment:
You do not need this `else`! You can just `return None` =)

Reply:
It would do the same, yes. The `else` is just there for readability.